hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/TraceStatsEnricher.java

```diff
@@ -0,0 +1,46 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import static org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants.HEAD_EVENT_ID;
+import static org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants.TOTAL_NUMBER_OF_UNIQUE_API_NODES;
+
+import java.util.List;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.ApiNode;
+import org.hypertrace.core.datamodel.shared.trace.AttributeValueCreator;
+import org.hypertrace.traceenricher.enrichment.AbstractTraceEnricher;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraph;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraphBuilder;
+
+public class TraceStatsEnricher extends AbstractTraceEnricher {
+
+  @Override
+  public void enrichTrace(StructuredTrace trace) {
+    ApiTraceGraph apiTraceGraph = ApiTraceGraphBuilder.buildGraph(trace);
+    if (apiTraceGraph.getApiNodeList().isEmpty()) {
+      return;
+    }
+    Event firstNodeHeadSpan = apiTraceGraph.getApiNodeList().get(0).getHeadEvent();
+    if (firstNodeHeadSpan == null) {
+      return;
+    }
+    addHeadSpanIdTraceAttribute(trace, firstNodeHeadSpan);
+    addTotalNumberOfUniqueApiNodesHeadSpanAttribute(
+        apiTraceGraph.getApiNodeList(), firstNodeHeadSpan);
+  }
+
+  private void addHeadSpanIdTraceAttribute(StructuredTrace trace, Event headSpan) {
+    AttributeValue attribute =
+        AttributeValue.newBuilder().setBinaryValue(headSpan.getEventId()).build();
+    trace.getAttributes().getAttributeMap().put(HEAD_EVENT_ID, attribute);
+  }
+
+  private void addTotalNumberOfUniqueApiNodesHeadSpanAttribute(
+      List<ApiNode<Event>> apiNodeList, Event headSpan) {
+    headSpan
```

Comment:
Do we want to enrich the span or the trace? 🤔 A head span having unique number of API nodes in the entire trace, sounds a bit weird

Reply:
Agree. This is a better fit for a trace attribute. If there are any downstream dependencies on the need for this to exist on the head span, we should be able to fix that.

Reply:
Same as previous comment, per original it should be in head span. If we want to debate this further lets use the slack channel I created and ask Ravi